### PR TITLE
New resource/data source: `azurerm_sentinel_alert_rule_ms_security_incident`

### PR DESCRIPTION
**Pull request by magodo**
This PR implement one of the alert rule resource required in #6448.

## Test Result

```bash
💤 😡 2 via 🦉 v1.14.2 make testacc TEST=./azurerm/internal/services/sentinel/tests TESTARGS="-run='TestAccAzureRMSentinelAlertRuleMsSecurityIncident_'"
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
TF_ACC=1 go test ./azurerm/internal/services/sentinel/tests -v -run='TestAccAzureRMSentinelAlertRuleMsSecurityIncident_' -timeout 180m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccAzureRMSentinelAlertRuleMsSecurityIncident_basic
=== PAUSE TestAccAzureRMSentinelAlertRuleMsSecurityIncident_basic
=== RUN   TestAccAzureRMSentinelAlertRuleMsSecurityIncident_complete
=== PAUSE TestAccAzureRMSentinelAlertRuleMsSecurityIncident_complete
=== RUN   TestAccAzureRMSentinelAlertRuleMsSecurityIncident_update
=== PAUSE TestAccAzureRMSentinelAlertRuleMsSecurityIncident_update
=== RUN   TestAccAzureRMSentinelAlertRuleMsSecurityIncident_requiresImport
=== PAUSE TestAccAzureRMSentinelAlertRuleMsSecurityIncident_requiresImport
=== CONT  TestAccAzureRMSentinelAlertRuleMsSecurityIncident_basic
=== CONT  TestAccAzureRMSentinelAlertRuleMsSecurityIncident_update
=== CONT  TestAccAzureRMSentinelAlertRuleMsSecurityIncident_requiresImport
=== CONT  TestAccAzureRMSentinelAlertRuleMsSecurityIncident_complete
--- PASS: TestAccAzureRMSentinelAlertRuleMsSecurityIncident_complete (151.06s)
--- PASS: TestAccAzureRMSentinelAlertRuleMsSecurityIncident_basic (214.84s)
--- PASS: TestAccAzureRMSentinelAlertRuleMsSecurityIncident_update (355.88s)
--- PASS: TestAccAzureRMSentinelAlertRuleMsSecurityIncident_requiresImport (366.84s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/sentinel/tests      366.876s

💤 😡 2 via 🦉 v1.14.2 make testacc TEST=./azurerm/internal/services/sentinel/tests TESTARGS="-run='TestAccDataSourceSentinelAlertRuleMsSecurityIncident_'"
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
TF_ACC=1 go test ./azurerm/internal/services/sentinel/tests -v -run='TestAccDataSourceSentinelAlertRuleMsSecurityIncident_' -timeout 180m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccDataSourceSentinelAlertRuleMsSecurityIncident_basic
=== PAUSE TestAccDataSourceSentinelAlertRuleMsSecurityIncident_basic
=== CONT  TestAccDataSourceSentinelAlertRuleMsSecurityIncident_basic
--- PASS: TestAccDataSourceSentinelAlertRuleMsSecurityIncident_basic (175.45s)
PASS
```

### CODE REVIEW

File: azurerm/internal/services/sentinel/resource_arm_sentinel_alert_rule_ms_security_incident.go [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+
+			"text_whitelist": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
```
[the remark this answers is not in the document]
Comment (magodo) [reply azurerm/internal/services/sentinel/resource_arm_sentinel_alert_rule_ms_security_incident.go:103]: Will add a `StringIsNotEmpty` as validation.

File: azurerm/internal/services/sentinel/parse/sentinel_alert_rules.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	ResourceGroup string
+	Workspace     string
+	Name          string
+}
+
+func SentinelAlertRuleID(input string) (*SentinelAlertRuleId, error) {
+	// Example ID: /subscriptions/<sub1>/resourceGroups/<grp1>/providers/Microsoft.OperationalInsights/workspaces/<workspace1>/providers/Microsoft.SecurityInsights/alertRules/<rule1>
+	groups := regexp.MustCompile(`^(.+)/providers/Microsoft\.SecurityInsights/alertRules/(.+)$`).FindStringSubmatch(input)
+	if len(groups) != 3 {
+		return nil, fmt.Errorf("faield to parse Sentinel Alert Rule ID: %q", input)
```
[the remark this answers is not in the document]
Comment (magodo) [reply azurerm/internal/services/sentinel/parse/sentinel_alert_rules.go:20]: Oops, thank you for pointing out!

File: website/docs/r/sentinel_alert_rule_ms_security_incident.html.markdown [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+* `display_name` - (Required) The friendly name of this Sentinel MS Security Incident Alert Rule.
+
+* `product_filter` - (Required) The Microsoft Security Service from where the alert will be generated. Possible values are `Azure Active Directory Identity Protection`, `Azure Advanced Threat Protection`, `Azure Security Center`, `Azure Security Center for IoT` and `Microsoft Cloud App Security`.
+
+* `severity_filter` - (Required) Only create incidents from alerts when alert severity level is contained in this list. Possible values are `High`, `Medium`, `Low` and `Informational`.
+
+~> **NOTE** At least one of the severity filters need to be set.
+
+---
```
[the remark this answers is not in the document]
Comment (magodo) [reply website/docs/r/sentinel_alert_rule_ms_security_incident.html.markdown:57]: This file is generated via `make scaffold-website`. This span line seems used to split required attributes and optional attributes. Do you mind check if this is really redundant? If so, then I think I can submit another PR to remove this span line from the tooling.

**Comment by magodo** [comment]
@mbfrahry Thank you for your comment! I have updated code accordingly, except for the change in document file, wanna get your confirmation.

## Test Result

```bash
💤 via 🦉 v1.14.2 make testacc TEST=./azurerm/internal/services/sentinel/tests TESTARGS="-run='TestAccAzureRMSentinelAlertRuleMsSecurityIncident_update'"
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
TF_ACC=1 go test ./azurerm/internal/services/sentinel/tests -v -run='TestAccAzureRMSentinelAlertRuleMsSecurityIncident_update' -timeout 180m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccAzureRMSentinelAlertRuleMsSecurityIncident_update
=== PAUSE TestAccAzureRMSentinelAlertRuleMsSecurityIncident_update
=== CONT  TestAccAzureRMSentinelAlertRuleMsSecurityIncident_update
--- PASS: TestAccAzureRMSentinelAlertRuleMsSecurityIncident_update (307.07s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/sentinel/tests      307.082s
```

**Comment by magodo** [comment]
I re-base my commits onto master branch to make the CI pass.

File: website/docs/r/sentinel_alert_rule_ms_security_incident.html.markdown (hunk 3 above)
[the remark this answers is not in the document]
Comment (mbfrahry) [reply website/docs/r/sentinel_alert_rule_ms_security_incident.html.markdown:57]: Ahhh, I see. We add the span line when defining what attributes go into blocks so I believe it should be removed here.

[the remark this answers is not in the document]
Comment (mbfrahry) [reply website/docs/r/sentinel_alert_rule_ms_security_incident.html.markdown:57]: Ah, after chatting internally, it seems like this is the way we're moving going forward. Apologies for calling it out!

**Pull request merged**